libs/contrib/Test/Golden.idr

```diff
@@ -0,0 +1,194 @@
+||| Core features required to perform Golden file testing.
+|||
+||| We provide the core functionality to run a *single* golden file test.
+||| This allows the developer freedom to design the rest of the test harness.
+|||
+||| This was originally used as part of Idris2's own test suite and
+||| the core functionality is useful for the many and not the few.
+||| Please see Idris2 test harness for example usage.
+|||
+||| # Test Structure
+|||
+||| This harness works from the assumption that each individual golden test comprises of a directory with the following structure:
+|||
+||| + `run.sh` a *shell* script that runs the test
+||| + `expected` the expected output of `run.sh`
+|||
+||| During testing, the test harness will generate an artefact named `output` and use `git diff` to examine the two files.
+||| This assumes that the expected has been checked into `git`.
+||| If `git diff` failes then the runner will simply present the expected and 'given' files.
+|||
+||| # Options
+|||
+||| The test harness has several options that must be set:
+|||
+||| + `idris2` the path of the executable we are testing.
+||| + `onlyNames` the list of tests to run relative to the generated executable.
+||| + `interactive` Should we update the expected file or not.
+|||
+||| # Usage
+|||
+||| When compiled to an executable the expected usage is:
+|||
+|||```sh
+|||runtests <path to executable> [--interactive] [--only [NAMES]]
```

Comment:
`<path to idris2>` maybe? The latest mention of an "executable" is talking
about `runtests` itself which confused me.

Reply:
changed to `path to executable under test`.